### PR TITLE
github action to release importnb

### PR DESCRIPTION
**Pull request by tonyfast**

### CODE REVIEW

**Comment by tonyfast** [comment]
@bollwyvl mind given a review on this pr. i'm not doing anything fancy with the version bumps. just wanted to check this approach was sane.

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+
+name: Python package
+on: push
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.8]
```
Comment (bollwyvl) [inline .github/workflows/release.yml:9]: doesn't even need a matrix... can just drop the `strategy` and use 3.8 (or 3.9) in the `python-version`

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+      run: |
+        python -m pip install --upgrade pip
+        pip install tox wheel
+        pip install .
+    - name: Test  
+      run: tox
+    - name: Build
+      run: |
+        python setup.py sdist bdist_wheel
+        python -m pip install --find-links=dist --no-index --ignore-installed --no-deps importnb
```
Comment (bollwyvl) [inline .github/workflows/release.yml:27]: no particular reason to install itself again (again again)...

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    - name: Publish package
+      if: github.event_name == 'push' && startsWith(github.event.ref, 'refs/tags')
```
Comment (bollwyvl) [inline .github/workflows/release.yml:29]: can probably hoist this up to on...

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -0,0 +1,33 @@
+
+name: Python package
+on: push
```
Comment (bollwyvl) [inline .github/workflows/release.yml:3]: If you don't mind clicking some UI buttons, maybe:

```yaml
on:
  release:
    types: [published]
```

File: .github/workflows/release.yml [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+      run: tox
+    - name: Build
+      run: |
+        python setup.py sdist bdist_wheel
+        python -m pip install --find-links=dist --no-index --ignore-installed --no-deps importnb
+    - name: Publish package
+      if: github.event_name == 'push' && startsWith(github.event.ref, 'refs/tags')
+      uses: pypa/gh-action-pypi-publish@master
+      with:
+        user: docfast
```
Comment (bollwyvl) [inline .github/workflows/release.yml:32]: we should really make a bot...

Comment (tonyfast) [reply .github/workflows/release.yml:32]: deathbot?

File: .github/workflows/release.yml (hunk 3 above)
Comment (tonyfast) [reply .github/workflows/release.yml:29]: can i skip this logic if i use the published trigger?

File: .github/workflows/release.yml [hunk 6]
```diff
@@ -1,17 +1,17 @@
...
-    strategy:
-      matrix:
-        python-version: [3.8]
+    python-version: 3.8
 
     steps:
     - uses: actions/checkout@v2
     - name: Set up Python ${{ matrix.python-version }}
-      uses: actions/setup-python@v1
+      uses: actions/setup-python@v2
```
Comment (bollwyvl) [inline .github/workflows/release.yml:14]: nice

File: .github/workflows/release.yml (hunk 3 above)
Comment (bollwyvl) [reply .github/workflows/release.yml]: yeah, that's the idea!

File: .github/workflows/release.yml (hunk 5 above)
Comment (bollwyvl) [reply .github/workflows/release.yml:30]: seems rather namesquatted already... perhaps deathbought or some other obviously intentional misspelling... or some entirely mortality-unrelated thing derived from some silly mechantronic beastie...

**Review by bollwyvl — APPROVED** [review]
won't know until it fails, but at least it won't publish anything, and we can probably try again! maybe do a `0.7.0rc0` to test after merging?

Also, we should probably put the version number in the release title, as the version is no longer clearly shown in the GH UI...

**Pull request merged**